File: scripts/add-descriptions.ts
/**
 * Add descriptions to all supplements.
 */
import { getDb } from "../src/lib/db";

const db = getDb();

const descriptions: Record<string, string> = {
  // Vitamins
  "Vitamin D": "Vitamin D is essential for calcium absorption, bone health, and immune function. Most people are deficient, especially those with limited sun exposure. Supplementation supports mood, hormone health, and reduces inflammation.",
  "Vitamin D3": "The most bioavailable form of vitamin D. Supports bone density, immune response, testosterone production, and mood regulation. Often paired with K2 for proper calcium metabolism.",
  "Vitamin K2 MK-7": "Directs calcium into bones and teeth rather than arteries. Works synergistically with vitamin D3. The MK-7 form has a longer half-life than MK-4, providing more sustained benefits.",
  "Multivitamin": "A comprehensive blend of essential vitamins and minerals to fill nutritional gaps in the diet. Supports overall health, energy metabolism, and immune function.",
  "Vitamin B-Complex": "B vitamins are crucial for energy production, brain function, and cell metabolism. Includes B1, B2, B3, B5, B6, B7, B9, and B12 — each playing distinct roles in converting food into energy.",

  // Minerals
  "Magnesium": "Involved in over 300 enzymatic reactions. Supports sleep quality, muscle relaxation, stress reduction, and heart health. Most people don't get enough from diet alone.",
  "Magnesium Glycinate": "A highly absorbable form of magnesium bound to glycine. Known for its calming effects — excellent for sleep, anxiety reduction, and muscle recovery without the laxative effect of other forms.",
  "Magnesium L-Threonate": "The only form of magnesium that effectively crosses the blood-brain barrier. Studied for cognitive enhancement, memory improvement, and synaptic plasticity. Often taken in the evening.",
  "Magnesium Threonate": "The only form of magnesium proven to significantly raise brain magnesium levels. Supports learning, memory, and cognitive function. Popularized by neuroscientists for its brain-specific benefits.",
  "Zinc": "Essential for immune function, testosterone production, wound healing, and protein synthesis. Commonly taken with magnesium (ZMA formula) for sleep and recovery.",
  "Electrolytes": "Minerals including sodium, potassium, magnesium, and calcium that regulate nerve function, muscle contraction, hydration, and pH balance. Critical during fasting, exercise, or low-carb diets.",
  "Sodium": "Essential electrolyte for fluid balance, nerve transmission, and muscle function. Often supplemented during fasting or low-carb diets when the body excretes more sodium.",

  // Omega-3s
  "Fish Oil": "Rich in EPA and DHA omega-3 fatty acids. Reduces inflammation, supports brain health, improves cardiovascular markers, and may enhance mood. Look for IFOS-certified products for purity.",
  "Omega-3": "Essential fatty acids (EPA and DHA) that the body cannot produce. Critical for brain structure, heart health, and reducing systemic inflammation. Most commonly sourced from fish oil or algae.",
  "Fish Oil / Omega-3": "High-quality omega-3 fatty acids support brain function, reduce inflammation, and protect cardiovascular health. EPA and DHA are the active forms — look for products with at least 1g combined EPA/DHA.",
  "Krill Oil": "Omega-3s from Antarctic krill, bound to phospholipids for potentially better absorption than standard fish oil. Also contains astaxanthin, a powerful antioxidant.",

  // Amino Acids
  "Creatine": "One of the most researched supplements in existence. Increases phosphocreatine stores for ATP regeneration — improving strength, power output, and muscle growth. Also shows cognitive benefits, particularly with sleep deprivation.",
  "Creatine Monohydrate": "The most studied and effective form of creatine. Supports muscle strength, power output, and exercise performance. Emerging research also shows cognitive benefits, particularly under sleep deprivation or mental fatigue.",
  "L-Theanine": "An amino acid found in green tea that promotes relaxation without sedation. Often paired with caffeine to enhance focus while reducing jitters. Supports alpha brain wave activity associated with calm alertness.",
  "L-Tyrosine": "A precursor to dopamine, norepinephrine, and adrenaline. Supports cognitive performance under stress, sleep deprivation, or cold exposure. Popular for focus and mental clarity.",
  "L-Glutamine": "The most abundant amino acid in the body. Supports gut health, immune function, and muscle recovery. Often used by athletes to reduce muscle soreness and by those with digestive issues.",
  "Glycine": "A calming amino acid that supports sleep quality, collagen production, and detoxification. Often paired with magnesium for enhanced sleep benefits. Also a precursor to glutathione.",
  "Collagen powder": "Provides the amino acids (glycine, proline, hydroxyproline) needed for skin elasticity, joint health, and connective tissue repair. Type I and III collagen support skin, hair, nails, and bones.",
  "BCAAs": "Branched-chain amino acids (leucine, isoleucine, valine) that support muscle protein synthesis and reduce exercise fatigue. Most beneficial when training fasted or on low protein diets.",
  "Beta-Alanine": "Increases muscle carnosine levels, which buffer acid buildup during high-intensity exercise. Improves performance in efforts lasting 1-4 minutes. Causes a harmless tingling sensation (paresthesia).",

  // Nootropics & Cognitive
  "Alpha-GPC": "A highly bioavailable choline source that increases acetylcholine levels in the brain. Supports focus, memory, and power output. Popular among those seeking cognitive enhancement and athletes.",
  "Caffeine": "The world's most widely consumed psychoactive substance. Blocks adenosine receptors to promote alertness, improve reaction time, and enhance exercise performance. Best cycled to prevent tolerance.",
  "Nicotinamide Riboside": "A NAD+ precursor that supports cellular energy production, DNA repair, and healthy aging. NR is one of the most efficient ways to boost NAD+ levels, which decline with age.",
  "NMN": "Nicotinamide mononucleotide — a direct precursor to NAD+, the coenzyme critical for cellular energy, DNA repair, and longevity pathways. NAD+ levels decline significantly with age.",
  "Resveratrol": "A polyphenol found in red wine and grapes. Activates sirtuins (longevity proteins) and provides antioxidant protection. Often paired with NMN for synergistic anti-aging effects.",
  "Nootropic Blend": "A combination of cognitive-enhancing compounds designed to improve focus, memory, processing speed, and mental clarity. Formulas vary but typically include choline sources, adaptogens, and amino acids.",
  "Alpha Brain": "A popular nootropic blend containing L-theanine, L-tyrosine, phosphatidylserine, and cat's claw. Promoted for focus, memory, and mental processing speed.",
  "Urolithin A": "A postbiotic that improves mitochondrial health by triggering mitophagy — the clearance of damaged mitochondria. Produced by gut bacteria from ellagitannins found in pomegranates.",

  // Antioxidants
  "Alpha-Lipoic Acid": "A potent antioxidant that is both water and fat soluble, allowing it to work throughout the body. Supports mitochondrial function, blood sugar regulation, and may help regenerate other antioxidants like vitamin C and glutathione.",
  "ALA": "Alpha-lipoic acid is a universal antioxidant that works in both water and fat-soluble environments. Supports glucose metabolism, nerve health, and mitochondrial function.",
  "CoQ10": "Coenzyme Q10 is essential for mitochondrial energy production. Levels decline with age and statin use. The ubiquinol form is more absorbable than ubiquinone.",
  "Curcumin": "The active compound in turmeric with powerful anti-inflammatory and antioxidant properties. Best absorbed when paired with piperine (black pepper) or formulated with lipid delivery systems.",
  "Sulforaphane / Broccoli Sprout Supplement": "A potent Nrf2 activator that upregulates the body's own antioxidant defenses. Derived from broccoli sprouts, which contain 20-50x more glucoraphanin (the precursor) than mature broccoli.",
  "Cocoa Extract / Flavanols": "Rich in epicatechin and other flavanols that improve blood flow, cognitive function, and cardiovascular health. Also supports nitric oxide production for better circulation.",
  "NAC": "N-Acetyl Cysteine is a precursor to glutathione, the body's master antioxidant. Supports liver detoxification, respiratory health, and may help with compulsive behaviors.",
  "NAC + Ginger + Curcumin": "A combination antioxidant and anti-inflammatory stack. NAC supports glutathione production, ginger aids digestion and reduces inflammation, and curcumin provides broad anti-inflammatory effects.",

  // Proteins & Greens
  "Collagen": "The most abundant protein in the body, providing structure to skin, joints, bones, and connective tissue. Supplementation supports skin elasticity, joint comfort, and gut health.",
  "Whey Protein": "A complete, fast-digesting protein rich in BCAAs. Ideal for muscle repair and growth post-workout. Also highly satiating for weight management.",
  "Protein Powder": "Convenient source of complete protein for muscle repair, satiety, and meeting daily protein targets. Types include whey, casein, plant-based blends, and collagen.",
  "Athletic Greens AG1": "A comprehensive greens powder with 75 vitamins, minerals, and whole-food-sourced ingredients. Includes probiotics, digestive enzymes, adaptogens, and superfoods.",
  "AG1": "Athletic Greens' flagship all-in-one supplement with 75 ingredients including vitamins, minerals, probiotics, digestive enzymes, and adaptogens. Designed to replace multiple supplements with one daily drink.",
  "Greens Powder": "A blend of dehydrated vegetables, grasses, algae, and sometimes probiotics and enzymes. Provides phytonutrients and alkalizing compounds to supplement a whole-foods diet.",
  "MCT Oil": "Medium-chain triglycerides that are rapidly absorbed and converted to ketones for quick energy. Popular in ketogenic diets and for cognitive performance. Derived from coconut or palm oil.",

  // Probiotics & Gut Health
  "Probiotic": "Beneficial bacteria that support gut microbiome health, digestion, and immune function. Different strains offer different benefits — look for multi-strain formulas with documented CFU counts.",
  "Digestive Enzymes": "Enzymes that help break down fats, proteins, and carbohydrates for better nutrient absorption. Particularly helpful for those with digestive issues or eating large meals.",
  "Fiber Supplement": "Dietary fiber supports digestive regularity, feeds beneficial gut bacteria, and helps regulate blood sugar and cholesterol. Most people consume less than the recommended 25-30g daily.",

  // Hormones & Adaptogens
  "Ashwagandha": "An adaptogenic herb that helps the body manage stress by modulating cortisol levels. Also supports thyroid function, testosterone levels, and sleep quality.",
  "Adaptogen Blend": "A combination of herbs (like ashwagandha, rhodiola, holy basil) that help the body adapt to physical, chemical, and biological stressors by normalizing physiological processes.",
  "Rhodiola Rosea": "An adaptogenic herb known for reducing fatigue, improving mental performance under stress, and supporting physical endurance. Particularly effective for burnout recovery.",
  "Melatonin": "The sleep hormone that regulates circadian rhythm. Supplementation can help with falling asleep faster, particularly for jet lag, shift work, or delayed sleep phase.",
  "DHEA": "A precursor hormone to testosterone and estrogen that declines with age. Sometimes supplemented for vitality and hormone optimization, though research on benefits is mixed.",

  // Other
  "Lutein + Zeaxanthin": "Carotenoids that accumulate in the retina and protect against blue light damage and age-related macular degeneration. Found naturally in leafy greens and egg yolks.",
};

const supplements = db.prepare("SELECT id, product_name, brand, category FROM supplements").all() as any[];

let updated = 0;
for (const s of supplements) {
  let desc = descriptions[s.product_name];

  // Try matching without brand-specific variants
  if (!desc) {
    for (const [key, val] of Object.entries(descriptions)) {
      if (s.product_name.toLowerCase().includes(key.toLowerCase()) || key.toLowerCase().includes(s.product_name.toLowerCase())) {
        desc = val;
        break;
      }
    }
  }

  if (desc) {
    db.prepare("UPDATE supplements SET description = ? WHERE id = ?").run(desc, s.id);
    updated++;
    console.log(`  ✓ ${s.product_name} (${s.brand})`);
  } else {
    console.log(`  ✗ ${s.product_name} (${s.brand}) — no match`);
  }
}

console.log(`\n✅ Updated ${updated} of ${supplements.length} supplements`);

File: scripts/add-descriptions.py
"""Add factual, educational descriptions to supplements missing them."""
import sqlite3
import os

DB_PATH = os.path.join(os.path.dirname(__file__), "..", "data", "directory.db")
db = sqlite3.connect(DB_PATH)

# Descriptions are educational — what the supplement is, why people take it.
# No medical claims, no dosage recommendations.
DESCRIPTIONS = {
    # ATHLEAN-X
    145: "ATHLEAN-Rx Whey Protein is a high-quality whey protein isolate designed to support muscle recovery and growth after training. Whey is a fast-digesting complete protein containing all essential amino acids. It is commonly used to meet daily protein targets, particularly around workouts.",
    146: "Creatine monohydrate is one of the most researched sports supplements. It helps replenish ATP, the primary energy currency of cells, supporting short bursts of high-intensity exercise like weightlifting and sprinting. ATHLEAN-Rx Creatine is a pure creatine monohydrate formulation.",
    147: "A comprehensive daily multivitamin from the ATHLEAN-X line, formulated to fill common micronutrient gaps in active individuals. Multivitamins provide a broad spectrum of vitamins and minerals that support overall health and training recovery.",

    # Ancient Nutrition
    152: "Bone broth protein powder derived from chicken bone broth, providing collagen, glucosamine, chondroitin, and amino acids. It offers a protein source that also supports joint, gut, and skin health, different from traditional whey or plant proteins.",
    151: "A blend of collagen proteins sourced from multiple animal types (bovine, chicken, fish, and eggshell membrane). Collagen is a structural protein important for skin elasticity, joint cushioning, and connective tissue integrity.",
    153: "A probiotic supplement providing beneficial bacteria intended to support digestive health and a balanced gut microbiome. Probiotics are live microorganisms that may aid digestion and immune function.",

    # Avmacol
    34: "Sulforaphane is a bioactive compound derived from broccoli sprouts, studied for its potential role in activating the body's Nrf2 antioxidant pathway. Avmacol provides a standardized source of glucoraphanin, the precursor to sulforaphane.",

    # Bare Performance Nutrition
    191: "Creatine monohydrate from Bare Performance Nutrition, a staple sports supplement used to support high-intensity exercise performance, muscle strength, and power output. It works by increasing phosphocreatine stores in muscle tissue.",
    193: "An electrolyte supplement designed to replenish sodium, potassium, and other minerals lost through sweat during training. Electrolytes are essential for hydration, muscle contraction, and nerve signaling.",
    192: "A pre-workout supplement formulated with ingredients intended to increase energy, focus, and blood flow during training. BPN Flight uses clinically-dosed ingredients like citrulline, beta-alanine, and caffeine.",
    190: "Whey protein powder from Bare Performance Nutrition, a complete protein source used to support muscle repair, recovery, and daily protein intake. Whey is rapidly absorbed and rich in branched-chain amino acids.",

    # Blueprint
    8: "Cocoa flavanols are plant compounds found in cocoa beans studied for their antioxidant properties and potential cardiovascular benefits. Blueprint's cocoa flavanols are part of Bryan Johnson's documented longevity protocol.",
    6: "Blueprint Essential Capsules are part of Bryan Johnson's comprehensive supplement protocol, designed to fill micronutrient gaps as part of a broader longevity-focused regimen. The specific formulation targets foundational vitamin and mineral needs.",
    18: "Blueprint Essential Capsules are part of Bryan Johnson's comprehensive supplement protocol, designed to fill micronutrient gaps as part of a broader longevity-focused regimen that includes dozens of daily supplements.",
    17: "Blueprint Longevity Mix is a custom-blended supplement powder from Bryan Johnson's Blueprint protocol, formulated to deliver multiple nutrients in a single serving as part of his documented anti-aging and health-optimization regimen.",

    # Bulletproof
    38: "Liposomal glutathione is formulated for enhanced absorption of glutathione, a tripeptide that functions as a major endogenous antioxidant in the body. Liposomal delivery encapsulates the glutathione to improve bioavailability.",
    39: "PQQ (Pyrroloquinoline Quinone) is a compound that supports mitochondrial function and biogenesis — the creation of new mitochondria in cells. It is often taken for its potential role in cellular energy production and neuroprotection.",

    # Doctor's Best
    156: "Glucosamine and chondroitin are compounds naturally found in joint cartilage, commonly supplemented for joint comfort and mobility. Doctor's Best formulation combines both in a single supplement.",

    # Dr. Berg's
    89: "Cod liver oil is a traditional supplement rich in omega-3 fatty acids (EPA and DHA), vitamin A, and vitamin D. It has been used for generations to support immune function, bone health, and cardiovascular wellness.",
    85: "A high-potassium electrolyte powder from Dr. Berg's product line, designed to support hydration and electrolyte balance. Potassium is a key electrolyte involved in nerve transmission, muscle contraction, and fluid regulation.",
    88: "Dr. Berg's Gallbladder Formula contains bile salts and digestive enzymes to support fat digestion and gallbladder function. Bile salts help emulsify dietary fats so they can be properly broken down and absorbed.",
    87: "Nutritional yeast is a deactivated yeast rich in B-complex vitamins, protein, and minerals. It has a savory, cheese-like flavor and is often used as a seasoning or supplement for its broad nutritional profile.",

    # Euvexia
    132: "Apple cider vinegar capsules provide acetic acid in a convenient form. Apple cider vinegar has been a popular traditional remedy and is commonly taken for its potential to support blood sugar regulation and digestion.",
    134: "A B-complex supplement providing the full spectrum of B vitamins, which are water-soluble and essential for energy metabolism, red blood cell formation, and neurological function. The B vitamins must be obtained regularly through diet or supplementation.",
    128: "Euvexia Electrolyte Powder is formulated to replenish key minerals — sodium, potassium, and magnesium — lost during physical activity and sweating. Proper electrolyte balance is critical for hydration status and muscle function.",
    130: "Magnesium citrate is a well-absorbed form of magnesium, an essential mineral involved in over 300 enzymatic reactions in the body. It is commonly taken for muscle relaxation, sleep support, and digestive regularity.",
    131: "Omega-3 fish oil provides the long-chain fatty acids EPA and DHA, which are important for cardiovascular health, brain function, and regulating inflammation. Fish oil is one of the most widely studied dietary supplements.",
    133: "A probiotic supplement providing strains of beneficial gut bacteria. Probiotics are live microorganisms intended to support and maintain a healthy digestive system and balanced gut flora.",
    129: "Vitamin D3 combined with vitamin K2 — a synergistic pairing where D3 supports calcium absorption and K2 helps direct calcium to bones rather than soft tissues. Both are fat-soluble vitamins important for bone and cardiovascular health.",

    # Fit Father Project
    166: "Fit Father Greens is a powdered greens supplement formulated to help busy men meet their daily vegetable and micronutrient intake. Greens powders typically blend leafy greens, vegetables, fruits, and herbs into a convenient drink.",
    167: "A daily multivitamin from the Fit Father Project designed for men over 40, formulated to address common nutrient gaps that may occur with aging. Multivitamins provide a broad foundation of vitamins and minerals.",
    168: "An omega-3 supplement from the Fit Father Project providing EPA and DHA fatty acids from fish oil. Omega-3s are important for heart health, brain function, and controlling inflammation, and are commonly recommended for middle-aged and older adults.",

    # Gorilla Mind
    68: "Gorilla Mind Smooth is a nootropic formulation designed to enhance cognitive performance, focus, and mental clarity. Nootropics are compounds taken to support various aspects of brain function including memory, attention, and processing speed.",
    67: "Gorilla Mode is a high-stimulant pre-workout formulation designed to increase energy, pump, and focus during training. It contains a blend of ingredients including citrulline, creatine, caffeine, and nootropic compounds.",
    174: "A pre-workout supplement from Gorilla Mind formulated to enhance training intensity, mental focus, and muscular endurance. Pre-workouts typically combine stimulants, amino acids, and performance-enhancing compounds.",

    # Gundry MD
    176: "Gundry MD Bio Complete 3 is a three-in-one formulation combining probiotics, prebiotics, and postbiotics designed to support gut health from multiple angles. It targets digestive comfort, regularity, and the gut microbiome.",
    177: "Gundry MD Energy Renew is a supplement formulated to support mitochondrial health and cellular energy production. It typically contains compounds like CoQ10, PQQ, and other nutrients involved in the body's energy pathways.",
    175: "Gundry MD Vital Reds is a polyphenol-rich blend of red fruits, berries, and botanical extracts formulated to support overall wellness. It provides a concentrated source of antioxidants from a wide variety of fruit sources.",
    178: "Vitamin D3 combined with vitamin K2 from Gundry MD. D3 supports calcium absorption and immune function, while K2 helps direct calcium to bones and teeth rather than accumulating in blood vessels.",

    # HLTH Code
    185: "HLTH Code Complete Meal is a meal replacement shake formulated with a science-backed ratio of protein, healthy fats, and fiber. It is designed to provide complete nutrition in a convenient format for busy individuals.",

    # Heart & Soil
    115: "Beef liver capsules are a desiccated organ supplement providing nutrients naturally concentrated in liver, including highly bioavailable vitamin A, B12, iron, and copper. Organ meats are among the most nutrient-dense foods available.",
    111: "Heart & Soil's Desiccated Organs is a blend of freeze-dried organ tissues from grass-fed sources, providing the unique nutrients found in various animal organs. This 'nose-to-tail' approach aims to deliver nutrients that are scarce in modern diets.",
    112: "Magnesium glycinate is a chelated form of magnesium bound to glycine, known for high bioavailability and gentle effect on digestion. It is commonly chosen for its calming properties and support for sleep quality.",
    113: "Vitamin D3 plus K2 from Heart & Soil. D3 is essential for calcium metabolism and immune function; K2 helps direct calcium to bones. Both are fat-soluble vitamins that work synergistically in the body.",

    # Jarrow Formulas
    28: "Methylfolate is the bioactive form of folate (vitamin B9), which does not require conversion by the body. It is important for DNA synthesis, red blood cell formation, and nervous system function. Methylfolate is preferred by individuals with MTHFR gene variants.",

    # Kion
    54: "Kion Aminos provides a blend of essential amino acids (EAAs) in free-form, designed for rapid absorption to support muscle protein synthesis, recovery, and energy during exercise. EAAs cannot be made by the body and must come from diet or supplementation.",
    53: "Kion Clean Energy Bar is a whole-food-based protein and energy bar formulated without artificial ingredients, designed to provide sustained energy from quality protein, healthy fats, and natural carbohydrates.",
    55: "Kion Flex is a joint support supplement formulated with ingredients like glucosamine, chondroitin, and turmeric to support joint comfort and mobility, particularly in active individuals.",

    # LMNT
    114: "LMNT Electrolyte Powder is a high-sodium electrolyte drink mix formulated for people following low-carb, keto, or active lifestyles. It provides sodium, potassium, and magnesium without sugar or artificial ingredients.",
    66: "An electrolyte supplement providing sodium, potassium, and magnesium in a science-backed ratio. Electrolytes are essential minerals that support hydration, muscle function, nerve signaling, and blood pressure regulation.",

    # Momentous
    15: "Apigenin is a bioactive flavonoid found in chamomile, parsley, and other plants. It has been studied for its potential to promote sleep quality and reduce anxiety by interacting with GABA receptors in the brain.",

    # NOW Foods
    170: "Magnesium citrate from NOW Foods, a widely available and well-absorbed form of magnesium. Magnesium supports muscle relaxation, nerve function, energy production, and sleep quality.",
    155: "Vitamin C (ascorbic acid) from NOW Foods. Vitamin C is a water-soluble antioxidant essential for immune function, collagen synthesis, and iron absorption. It cannot be produced by the human body and must be obtained from diet.",

    # Nordic Naturals
    202: "Omega-3 algae oil provides EPA and DHA from a sustainable, plant-based source — making it suitable for vegetarians and vegans. Algae oil offers the same omega-3 benefits as fish oil without the fish.",

    # Outwork Nutrition
    117: "Creatine monohydrate from Outwork Nutrition, a pure form of one of the most extensively studied sports supplements. Creatine supports ATP regeneration during high-intensity exercise, improving strength, power, and muscle recovery.",
    116: "Outwork Pre-Workout is formulated to increase energy, mental focus, and muscular endurance before training. It combines stimulants, amino acids, and performance-enhancing ingredients for an effective training session.",
    121: "Outwork Sleep+ Recovery is formulated to support restful sleep and overnight recovery. Quality sleep is essential for muscle repair, hormone regulation, and cognitive function.",
    118: "Whey protein isolate from Outwork Nutrition, a fast-digesting complete protein that supports muscle repair and recovery after training. Whey isolate is filtered to contain minimal fat and lactose.",

    # Prescription
    50: "Metformin is a prescription medication primarily used to manage type 2 diabetes by improving insulin sensitivity and reducing liver glucose production. Some longevity researchers study it for its potential effects on aging pathways, but it is not approved for this use.",

    # Pure Encapsulations
    204: "Omega-3 fish oil from Pure Encapsulations providing EPA and DHA. Pure Encapsulations products are formulated without unnecessary additives and undergo rigorous testing for quality and purity.",

    # RP Strength
    127: "Magnesium glycinate from RP Strength, a highly bioavailable form of magnesium bound to glycine. This form is commonly chosen for its calming effects on the nervous system and support for quality sleep.",
    126: "Omega-3 fish oil from RP Strength providing EPA and DHA fatty acids to support cardiovascular health, brain function, and a healthy inflammatory response.",
    122: "RP Creatine Monohydrate is a pure creatine supplement from Renaissance Periodization, a company focused on evidence-based sports nutrition. Creatine supports strength, power output, and lean body mass when combined with resistance training.",
    124: "RP Pre-Workout is formulated to enhance training performance through ingredients that support energy production, focus, and muscular endurance during workouts.",
    123: "RP Whey Protein from Renaissance Periodization, a high-quality protein source designed to support muscle recovery, repair, and daily protein targets for athletes and fitness enthusiasts.",
    125: "Vitamin D3 from RP Strength. Vitamin D is a fat-soluble vitamin essential for calcium absorption, bone health, and immune system function. Many people have insufficient levels, especially in winter months.",

    # Thorne
    135: "Thorne Research Multivitamin is a comprehensive daily nutrient supplement known for its high-quality ingredient sourcing and formulation standards. It provides a broad spectrum of essential vitamins and minerals in bioavailable forms.",
    120: "Omega-3 fish oil from Thorne Research, a well-regarded supplement brand that supplies many professional sports teams. Their fish oil provides EPA and DHA fatty acids and is tested for purity and potency.",

    # Various
    161: "An electrolyte blend providing sodium, potassium, and magnesium — the three primary electrolytes lost through sweat. Maintaining electrolyte balance is important for hydration, muscle contraction, nerve function, and preventing cramps.",
    162: "Vitamin D3 is a fat-soluble vitamin synthesized in skin upon sun exposure and obtained through certain foods. It plays a critical role in calcium absorption, bone density, immune function, and mood regulation.",

    # Various (self-assembled)
    30: "A custom-blended micronutrient smoothie assembled from various individual ingredients, tailored to personal nutritional needs. This approach allows for personalized dosing and ingredient selection based on individual goals.",
}

for supp_id, desc in DESCRIPTIONS.items():
    db.execute("UPDATE supplements SET description = ? WHERE id = ?", (desc, supp_id))

db.commit()
updated = len(DESCRIPTIONS)
print(f"Updated {updated} supplements with descriptions")

# Verify
missing = db.execute("SELECT COUNT(*) FROM supplements WHERE description IS NULL OR description = ''").fetchone()[0]
print(f"Still missing: {missing}")

db.close()
